internal/command/init.go

```diff
@@ -199,11 +216,28 @@ the new provisioners will take precedence.
 
 		if v, _ := cmd.Flags().GetStringArray(initCmdProvisionerFlag); len(v) > 0 {
 			for i, vi := range v {
+				var data []byte
+
+				if vi == "-" {
+					data, err = GetStdinFile(cmd.Context())
+				} else {
+					// Existing URI loading logic
+					data, err = uriget.GetFile(cmd.Context(), vi)
+				}
+
 				data, err := uriget.GetFile(cmd.Context(), vi)
 				if err != nil {
 					return fmt.Errorf("failed to load provisioner %d: %w", i+1, err)
 				}
-				if err := loader.SaveProvisionerToDirectory(sd.Path, vi, data); err != nil {
+
+				var saveFilename string
+				if vi == "-" {
+					saveFilename = fmt.Sprintf("stdin-provisioner-%d%s", i+1, loader.DefaultSuffix)
```

Comment:
The file name here needs to look like a provisioners file. I'd also suggest dropping the `%d` suffix since we can only ever have one stdin file.
```suggestion
					saveFilename = "from-stdin.provisioners.yaml"
```

Reply:
sure @astromechza ill make changes right now !